lib/kracken/controllers/authenticatable.rb

```diff
@@ -124,6 +112,30 @@ def user_signed_in?
 
       private
 
+      def session_present?
+        session[:user_id] && session[:user_cache_key]
```

Comment:
Can we add a note stating we are intentionally not checking `session[:user_uid]` here to handle the existing session upgrade case? Or am I miss understanding this check?